### ISSUE TITLE
Improve auxiliary keys in header menu

### ISSUE THREAD

**Issue by kacperknapik** [SAP/spartacus#6460]
### Problem
Currently when you tab through the header menu elements, and you want to open dropdown menu, you have to press space - this opens up the dropdown. But after tabbing further, to another nav element for example, and you go back, dropdown is automagically opened once again, even though it should be hidden (should go back to the initial state - which is closed).

### Solution
Force dropdowns to go back to the initial state (being closed) after tabbing to the next nav element.

**Comment by developpeurweb** [comment]
In this page there are some examples for inspiration: https://cxwiki.sap.com/display/spar/Key+Interactions+for+Category+Navigation+Menus

**Comment by mirkosoft** [comment]
Talk with @tobi-or-not-tobi first in order to make use of "swiss army knife" he created for a11y focus things.
